### PR TITLE
Add completions for direnv

### PR DESCRIPTION
**Pull request by xanderio**
## Description

Talk about your changes here.

Fixes issue #

## TODOs:

- [ ] Changes to fish usage are reflected in user documentation/manpages.
- [ ] Tests have been added for regressions fixed
- [ ] User-visible changes noted in CHANGELOG.rst

### CODE REVIEW

File: share/completions/direnv.fish [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+complete -c direnv -n "not __fish_seen_subcommand_from $commands" \
+    -a "allow permit grant" -d "Grants direnv permission to load the given .envrc or .env file"
+complete -c direnv -n "not __fish_seen_subcommand_from $commands" \
+    -a "block deny revoke" -d "Revokes the authorization of a given .envrc or .env file"
+complete -c direnv -n "not __fish_seen_subcommand_from $commands" \
+    -a edit -d "Opens PATH_TO_RC or the current .envrc or .env into an \$EDITOR and allow the file to be loaded afterwards"
+complete -c direnv -n "not __fish_seen_subcommand_from $commands" \
+    -a exec -d "Executes a command after loading the first .envrc or .env found in DIR"
+complete -c direnv -n "not __fish_seen_subcommand_from $commands" \
+    -a fetchurl -d "Fetches a given URL into direnv's CAS"
```
Comment (faho) [inline share/completions/direnv.fish:16]: What's "CAS" in this context?

File: share/completions/direnv.fish [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+complete -c direnv -n "not __fish_seen_subcommand_from $commands" \
+    -a prune -d "removes old allowed files"
+complete -c direnv -n "not __fish_seen_subcommand_from $commands" \
+    -a reload -d "triggers an env reload"
+complete -c direnv -n "not __fish_seen_subcommand_from $commands" \
+    -a status -d "prints some debug status information"
+complete -c direnv -n "not __fish_seen_subcommand_from $commands" \
+    -a stdlib -d "Displays the stdlib available in the .envrc execution context"
+complete -c direnv -n "not __fish_seen_subcommand_from $commands" \
+    -a version -d "prints the version or checks that direnv is older than VERSION_AT_LEAST"
```
Comment (faho) [inline share/completions/direnv.fish:30]: Is this really *older than*? I would expect it to be "newer than", so you can do things like

```fish
direnv --version 1.2
```

to confirm that you are running *at least* version 1.2 and can therefore do version 1.2 things.

Comment (krobelus) [reply share/completions/direnv.fish:30]: yeah that's misleading wording by `direnv --help`, how about `[..] checks that direnv is VERSION_AT_LEAST or newer`

File: share/completions/direnv.fish (hunk 1 above)
Comment (xanderio) [reply share/completions/direnv.fish:16]: This wording is directly copied from `direnv --help` and even after looking at the source code I have no concrete answer to this. It probably stands for **C**ontent **A**ddressable **S**torage, but that's just an adjudicated guess.

File: share/completions/direnv.fish (hunk 2 above)
Comment (xanderio) [reply share/completions/direnv.fish:30]:
```suggestion
    -a version -d "prints the version or checks that direnv is VERSION_AT_LEAST or newer"
```

**Comment by faho** [comment]
Merged, thanks!

File: share/completions/direnv.fish (hunk 1 above)
Comment (faho) [reply share/completions/direnv.fish:16]: Tbh I'd suggest to upstream to rephrase that. Until then it's good enough for us.

**Pull request merged**